src/databases/Silo/avtSiloFileFormat.C

```diff
@@ -90,6 +90,16 @@
 #include <string>
 #include <vector>
 
+#ifndef DB_ZONETYPE_QUAD_BEAM
+// Define fake elts for avoiding ifdef all the time
+#define DB_ZONETYPE_QUAD_BEAM      -1
```

Comment:
I like the goal here...to simplify the repeated use of `#ifndef...#endif` logic blocks. But, I am uneasy about the definitions to arbitrary negative values. Since we wind up enumerating each missing type on its own line here, why not use the values from `silo.h.in`? Or, use the *linear* counterparts (e.g. `#define DB_ZONETYPE_QUAD_BEAM DB_ZONETYPE_BEAM`) from `silo.h.in`? Generally speaking, once a value such of this is defined in Silo, we will *never* change it. The value is persisted to files and so should never be changed.

Reply:
Of course ! You are right, I'm going to take the values from silo.h.in, so that even a VisIt compiled against an old version of Silo will have the feature ! I will make a PR for that soon.